CMP/Setup/SetupCMP/Helpers/InstallActionProcessor.cs

```diff
@@ -151,52 +151,48 @@ in PropertyBagDictionary.Instance.GetProperty<ArrayList>(PropertyBagConstants.It
             //Registering the resource provider
             try
             {
-                string errStr = "";
+                bool isRpSuccess = true;
                 string RpFile = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location) + SetupConstants.RPFilename;
-                if (returnValue == SetupReturnValues.Successful && isUninstall != true && SetupConstants.WapDBOnRemoteServer == false)
+              
+                if (returnValue == SetupReturnValues.Successful && isUninstall != true )
                 {
-
-                    Process p = new Process();
-                    p.StartInfo.FileName = "Powershell.exe";
-                    p.StartInfo.Arguments = RpFile;
-                    p.StartInfo.UseShellExecute = false;
-                    p.StartInfo.RedirectStandardInput = true;
-                    p.StartInfo.RedirectStandardOutput = true;
-                    p.StartInfo.RedirectStandardError = true;
-                    p.StartInfo.CreateNoWindow = true;
-                    p.Start();
-                    errStr = p.StandardError.ReadToEnd();
-                    if (errStr != "") SetupLogger.LogError("Error in registering resource provider - " + errStr);
-                    p.Close();
-                    p.Dispose();
+                    WSManConnectionInfo connectionInfo = new WSManConnectionInfo();
+                    string computerName = "";
+                    if (SetupConstants.WapDBOnRemoteServer)
+                      computerName = SetupDatabaseHelper.GetAdminApiMachineName(true);
+                    else
+                      computerName=  Environment.MachineName;
+                    if (computerName != "")
+                    {
+                        connectionInfo.ComputerName = computerName;
+                        Runspace runspace = RunspaceFactory.CreateRunspace(connectionInfo);
+                        runspace.Open();
+                        StreamReader sr = new StreamReader(RpFile);
+                        string script = sr.ReadToEnd();
+                        using (PowerShell ps = PowerShell.Create())
+                        {
+                            ps.Runspace = runspace;
+                            ps.AddScript(script);
+                            var results = ps.Invoke();
+                            if (ps.Streams.Error.Count > 0)
+                            {
+                                isRpSuccess = false;
+                                SetupLogger.LogError("Error in registering resource provider - " + ps.Streams.Error.Count);

```

Comment:
Minor detail, but I would just specify that the resource provider being registered is the Azure Pack Connector (or Phoenix) to be more precise with the message.